### PR TITLE
Upgrading Graal.js to 20.3 LTS version

### PR DESCRIPTION
**Pull request by jtulach**
There is new LTS version of GraalVM - 20.3.0. There is also new latest version 21.0.0. Let's upgrade our gates to test against these. Also, let's update our code to build and use the LTS binaries when running on regular JDK.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
The 50a9059 change workaround the test error in Nashorn compatibility mode. However now we have [similar failure](https://travis-ci.com/github/apache/netbeans/jobs/479351728) when using `GraalVM:js` engine - that one is supposed to work, but it doesn't. Reported as _GR-29073_ against GraalVM 20.3.0.

File: webcommon/libs.graaljs/manifest.mf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Manifest-Version: 1.0
 AutoUpdate-Show-In-Client: false
-OpenIDE-Module: org.netbeans.libs.graaljs/1
+OpenIDE-Module: org.netbeans.libs.graaljs/2
```
Comment (sdedic) [inline webcommon/libs.graaljs/manifest.mf:3]: Why new major version ?

Comment (JaroslavTulach) [reply webcommon/libs.graaljs/manifest.mf:3]: There were incompatible changes in the `IS_MODULE_BODY` & co. values.

**Review by sdedic — APPROVED** [review]

**Comment by JaroslavTulach** [comment]
Integration of this PR has to wait, there is a bug (internal number GR-29073) that causes failures on GraalVM 20.3 and 21.0:
```
 com.oracle.truffle.api.dsl.UnsupportedSpecializationException: Unexpected values provided for IsCallableNodeGen@58f174d9: [com.oracle.truffle.js.runtime.builtins.JSArray$ArrayLengthProxyProperty@5be82d43], [ArrayLengthProxyProperty]
  org.graalvm.polyglot.PolyglotException: com.oracle.truffle.api.dsl.UnsupportedSpecializationException: Unexpected values provided for IsCallableNodeGen@58f174d9: [com.oracle.truffle.js.runtime.builtins.JSArray$ArrayLengthProxyProperty@5be82d43], [ArrayLengthProxyProperty]
  	at com.oracle.truffle.js.nodes.unary.IsCallableNodeGen.executeAndSpecialize(IsCallableNodeGen.java:268)
  	at com.oracle.truffle.js.nodes.unary.IsCallableNodeGen.executeBoolean(IsCallableNodeGen.java:109)
  	at com.oracle.truffle.js.nodes.interop.KeyInfoNode.cachedOwnProperty(KeyInfoNode.java:113)
```
the bug has already been fixed and the fix will be part of GraalVM 21.1. The fix will hopefully also be backported to some 20.3 update to make sure NetBeans works fine with the latest GraalVM LTS release.

**Pull request merged**